### PR TITLE
_to_copy decomp

### PR DESCRIPTION
**Pull request by ngimel**
Per title

@diff-train-skip-merge

### CODE REVIEW

**Review by Chillee — APPROVED** [review]
lol

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1000,6 +1000,34 @@ def _fused_dropout_decomposition(input, p, generator=None):
...
+    if device is not None and device != x.get_device():
+        # avoid conversions on cpu
+        if dtype is not None and device.type == "cpu":
+            x = torch._prims.convert_element_type(x, dtype)
+            dtype_converted = True
+        x = torch._prims.device_put(x, device)
+    if dtype and not dtype_converted:
+        x = torch._prims.convert_element_type(x, dtype)
+    if memory_format:  # no ref/prim for memory format
+        x = x.to(memory_format=memory_format)
```
Comment (Chillee) [inline torch/_decomp/decompositions.py:1027]: I'm somewhat worried this is gonna be recursive 🤔

Comment (ngimel) [reply torch/_decomp/decompositions.py]: OpInfo dtype conversions tests pass

Comment (IvanYashchuk) [reply torch/_decomp/decompositions.py]: And how is this function appearing in the dtype conversions tests? Can you please give a specific command to launch a test that uses this implementation?

Comment (ngimel) [reply torch/_decomp/decompositions.py]: `python test/test_decomp.py -v -k test_comprehensive_short_cpu_float16`

Comment (IvanYashchuk) [reply torch/_decomp/decompositions.py]: Thank you! Now I know about `.short()` and other dtype methods OpInfos.

**Pull request closed without merging**